src/compiler.ts

```diff
@@ -10026,6 +10026,95 @@ export class Compiler extends DiagnosticEmitter {
   // === Specialized code generation ==============================================================
 
   /** Makes a constant zero of the specified type. */
+  /** Checks if an expression evaluates to a zero value for the given type. */
+  shouldSkipZeroInit(expr: ExpressionRef, type: Type): bool {
+    let module = this.module;
+    // Try to evaluate the expression at compile time
+    let evaled = module.runExpression(expr, ExpressionRunnerFlags.Default);
```

Comment:
`runExpression` and check result is not expected behavior in debug mode.
consider `x = A - A`. In debug mode, we should emit `(i32.sub (global.get A) (global.get A))` instead of simplify it to `(i32.const 0)` even totally remove it.